rules/framework.bzl

```diff
@@ -340,20 +355,87 @@ def _apple_framework_packaging_impl(ctx):
             swift_common.create_module(name = swiftmodule_name, swift = swift_module),
         ]
 
-    # Eventually we need to remove any reference to objc provider
-    # and use CcInfo instead, see this issue for more details: https://github.com/bazelbuild/bazel/issues/10674
-    objc_provider = apple_common.new_objc_provider(**objc_provider_fields)
+    direct_headers = []
+
+    if virtualize_frameworks:
+        import_vfsoverlays = []
+        for dep in ctx.attr.vfs:
+            if not VFSOverlayInfo in dep:
```

Comment:
what's the situation where a dep of attr.vfs is not a VFSOverlayInfo?

Reply:
You'll find this in the main callsite of `framework_packing` - there's a few edges around imported frameworks and we pass this in ATM